### PR TITLE
Change guideline for API versioning

### LINKED ISSUE #310: Guideline on API Version Number is Impractical for Reviews
The current guideline for API version numbers is impractical for the review process. It requires API writers to update the minor or major version after the review has been done, while changing the "review" number on each request (see [rule 116](https://zalando.github.io/restful-api-guidelines/#116)).

This results in the problem that a reviewer cannot review the version number as it will be only changed afterwards. This results in more errors of releasing API with incorrect versions. Today most reviews are using GH/GHE which provide a commit hash for distinguishing reviewed API versions. In addition, this rule is hardly unknown or in consequences is unexpected, so that it is mostly not applied correctly.

I suggest to change the rule to standard semantic versioning, which is the most expected and formally already used process for API versioning.

### PR DESCRIPTION
**Pull request by tkrop**
Fixes #310.

### CODE REVIEW

File: chapters/compatibility.adoc [hunk 1]
```diff
@@ -289,9 +289,32 @@ versioning, and go instead with media type versioning and content
 negotiation (see above).
 
 [#116]
-== {SHOULD} Provide Version Information in OpenAPI Documentation
+== {MUST} Use Semantic Versioning
 
-Only the documentation, not the API itself, needs version information.
+OpenAPI allows to specify the an API specification version in
```
Comment (maxim-tschumak) [inline chapters/compatibility.adoc]: the an API version ->
the ~an~ API version

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by whiskeysierra** [comment]
:+1:

**Comment by whiskeysierra** [comment]
@tkrop Can you resolve the conflict?

**Comment by tkrop** [comment]
👍

**Comment by whiskeysierra** [comment]
:+1:

**Pull request merged**